types/src/validator_verifier.rs

```diff
@@ -275,7 +290,38 @@ impl ValidatorVerifier {
 
         multi_sig
             .verify(message, &aggregated_key)
-            .map_err(|_| VerifyError::InvalidSignature)?;
+            .map_err(|_| VerifyError::InvalidMultiSignature)?;
+        Ok(())
+    }
+
+    pub fn verify_aggregated_signatures<T: CryptoHash + Serialize>(
+        &self,
+        messages: &[&T],
+        aggregated_signature: &AggregateSignature,
+    ) -> std::result::Result<(), VerifyError> {
+        // Verify the number of signature is not greater than expected.
+        Self::check_num_of_voters(self.len() as u16, aggregated_signature.get_voters_bitvec())?;
+        let mut pub_keys = vec![];
+        let mut authors = vec![];
+        for index in aggregated_signature.get_voters_bitvec().iter_ones() {
+            let validator = self
+                .validator_infos
+                .get(index)
+                .ok_or(VerifyError::UnknownAuthor)?;
+            authors.push(validator.address);
+            pub_keys.push(validator.public_key());
+        }
+        // Verify the quorum voting power of the authors
+        self.check_voting_power(authors.iter())?;
```

Comment:
I tried that but realized it's not worth it and makes the code more confusing and complex. Because in case of last few verification steps like generating the aggregated pub key and validating the message is different.